engine/common/requester/engine.go

```diff
@@ -234,18 +237,24 @@ func (e *Engine) addEntityRequest(entityID flow.Identifier, selector flow.Identi
 // Force will force the requester engine to dispatch all currently
 // valid batch requests.
 func (e *Engine) Force() {
-	count := uint(0)
-	for {
```

Comment:
[Force is called](https://github.com/onflow/flow-go/blob/master/engine/execution/ingestion/engine.go#L988) by ingestion engine when requesting collections.

However, this Force method is blocking, which causes the call `matchOrRequestCollections` to be blocking, which created other side effects.